main/src/library/Fixpoint3/UniqueInts.flix

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2025 Casper Dalgaard Nielsen
+ *                Adam Yasser Tallouzi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+///
+/// The purpuse of `UniqueInts` is to assign unique numbers/identifiers to elements.
+///
+/// A map from the assigned identifiers to the original values is also provided.
+///
+
+mod Fixpoint3 {
+    @Internal
+    pub enum UniqueInts[k: Type, r: Region] {
```

Comment:
I think I would use a struct here too

Reply:
In this case it would be a pure struct, which we never added

Reply:
pure structs! 🙌 pure structs! 🙌 pure structs! 🙌